backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java

```diff
@@ -59,6 +59,30 @@
       nativeQuery = true)
   public int getTestResultsCount(UUID facilityId);
 
+  @Query(
+      value =
+          "WITH FILTEREDEVENTS AS ("
```

Comment:
Hmmm, yeah I think moving that first part into a constant is a good idea. IIRC the reason the optional `patientId` didn't work is because Hibernate refused to accept a null value for `patientId`, even if I inserted a null check into the query itself, ie `WHERE (patientId IS NOT NULL && patientId = ?)`